modelskill/model/dfsu.py

```diff
@@ -158,27 +157,27 @@ def _extract_point(
         assert method in ["nearest", "contained", "inverse_distance"]
         n_nearest = 5 if method == "inverse_distance" else 1
 
-        assert isinstance(
-            self.data, (mikeio.dfsu.Dfsu2DH, mikeio.DataArray, mikeio.Dataset)
-        )
-
-        x, y = observation.x, observation.y
+        x, y, z = observation.x, observation.y, observation.z
         if not self._in_domain(x, y):
             raise ValueError(
                 f"PointObservation '{observation.name}' ({x}, {y}) outside model domain!"
             )
 
         if method == "contained":
-            xy = np.atleast_2d([x, y])
-            elemids = self.data.geometry.find_index(coords=xy)
-            if isinstance(self.data, mikeio.dfsu.Dfsu2DH):
-                ds_model = self.data.read(elements=elemids, items=self.sel_items.all)
-            elif isinstance(self.data, mikeio.Dataset):
+            if z is None:
+                elemids = self.data.geometry.find_index(x=x, y=y)
+            else:
+                elemids = self.data.geometry.find_index(x=x, y=y, z=z)
```

Comment:
What if data is a dfsu2d file here? I guess some users would write z=0 or z=np.nan in their observations (or maybe even the water depth 🙄) when extracting from 2d dfsu files. 

We could check the geometry before providing z

Reply:
We could, but we don't want to be tightly coupled to the actual the types in MIKE IO.

So if we we check that the geometry supports extraction of z, without explicit types, that would be great.